numpy/f2py/_isocbind.py

```diff
@@ -13,48 +13,115 @@
+        'c_long_double': 'long double'
     },
     'complex': {
-        'c_float_complex': 'complex_float',
-        'c_double_complex': 'complex_double',
-        'c_long_double_complex': 'complex_long_double'
+        'c_float_complex': 'float _Complex',
+        'c_double_complex': 'double _Complex',
+        'c_long_double_complex': 'long double _Complex'
     },
     'logical': {
-        'c_bool': 'unsigned_char'  # _Bool <=> 'unsigned_char' for now
+        'c_bool': '_Bool'
     },
     'character': {
         'c_char': 'char'
     }
 }
 
-# TODO: See gh-25229
-isoc_c2pycode_map = {}
-iso_c2py_map = {}
+# TODO: At some point these should be included, but then they'd need special
+# handling in cfuncs.py e.g. needs[int64_t_from_pyobj] These are not very hard
+# to add, since they all derive from the base `int_from_pyobj`, e.g. the way
+# `short_from_pyobj` and others do
+isoc_c2pycode_map = {
+    'int': 'i',  # int
+    'short int': 'h',  # short int
+    'long': 'l',  # long int
+    'long long': 'q',  # long long int
+    'signed char': 'b',  # signed char
+    'size_t': 'I',  # size_t (approx unsigned int)
+    'int8_t': 'b',  # int8_t
+    'int16_t': 'h',  # int16_t
+    'int32_t': 'i',  # int32_t
+    'int64_t': 'q',  # int64_t
+    'int_least8_t': 'b',  # int_least8_t
+    'int_least16_t': 'h',  # int_least16_t
+    'int_least32_t': 'i',  # int_least32_t
+    'int_least64_t': 'q',  # int_least64_t
+    'int_fast8_t': 'b',  # int_fast8_t
+    'int_fast16_t': 'h',  # int_fast16_t
+    'int_fast32_t': 'i',  # int_fast32_t
+    'int_fast64_t': 'q',  # int_fast64_t
+    'intmax_t': 'q',  # intmax_t (approx long long)
+    'intptr_t': 'q',  # intptr_t (approx long long)
+    'ptrdiff_t': 'q',  # intptr_t (approx long long)
+    'float': 'f',  # float
+    'double': 'd',  # double
+    'long double': 'g',  # long double
+    'float _Complex': 'F',  # float  _Complex
+    'double _Complex': 'D',  # double  _Complex
+    'long double _Complex': 'D',  # very approximate complex
+    '_Bool': 'i',  #  Bool but not really
+    'char': 'c',   # char
+}
+
+iso_c2py_map = {
+    'int': 'int',
+    'short int': 'int',                 # forced casting
+    'long': 'int',
+    'long long': 'long',
+    'signed char': 'int',           # forced casting
+    'size_t': 'int',                # approx Python int
+    'int8_t': 'int',                # forced casting
+    'int16_t': 'int',               # forced casting
+    'int32_t': 'int',
+    'int64_t': 'long',
+    'int_least8_t': 'int',          # forced casting
+    'int_least16_t': 'int',         # forced casting
+    'int_least32_t': 'int',
+    'int_least64_t': 'long',
+    'int_fast8_t': 'int',           # forced casting
+    'int_fast16_t': 'int',          # forced casting
+    'int_fast32_t': 'int',
+    'int_fast64_t': 'long',
+    'intmax_t': 'long',
+    'intptr_t': 'long',
+    'ptrdiff_t': 'long',
+    'float': 'float',
+    'double': 'float',              # forced casting
```

Comment:
```suggestion
    'double': 'double',              # forced casting
```
@Pranavchiku 
This is where the mapping is done to the special handlers in `C`, i.e. where it is decided if one needs to call `float_from_pyobj` or `double_from_pyobj`.

Reply:
Okay, perfect. I was wondering the same before updating assert statement for float.